checks.d/vsphere.py

```diff
@@ -692,17 +661,15 @@ def _cache_morlist_raw(self, instance):
 
         i_key = self._instance_key(instance)
         self.log.debug("Caching the morlist for vcenter instance %s" % i_key)
-        if i_key in self.morlist_raw and len(self.morlist_raw[i_key]) > 0:
-            self.log.debug(
-                "Skipping morlist collection now, RAW results "
-                "processing not over (latest refresh was {0}s ago)".format(
-                    time.time() - self.cache_times[i_key][MORLIST][LAST])
-            )
-            return
-        self.morlist_raw[i_key] = []
-
-        server_instance = self._get_server_instance(instance)
-        root_folder = server_instance.content.rootFolder
+        for resource_type in RESOURCE_TYPE_MAP:
```

Comment:
Well, I did that because now we pop() the elements per resource type in `_cache_morlist_process()`, before we knew when the list was empty, we were done. Now we may have processed all ComputeResources, or whatever, but not the VM's, so just checking for the length of the dictionary wouldn't be enough. Besides, we pop from the lists, not the keys.  `self.morlist_raw[i_key][resource_type]` dicts are created the first time around, but they then exist through out, I believe, so the `len(self.morlist_raw[i_key]) == len(RESOURCE_TYPE_MAP)` in subsequent iterations.